GitHubReferences.ps1

```diff
@@ -0,0 +1,219 @@
+    }
+
+    if ($OwnerName -xor $RepositoryName)
+    {
+        $message = 'You must specify both Owner Name and Repository Name.'
+        Write-Log -Message $message -Level Error
+        throw $message
+    }
+    $uriFragment = "/repos/$OwnerName/$RepositoryName/git/refs/$Reference"
+    $description =  "Getting Reference $Reference for $RepositoryName"
+
+    $params = @{
+        'UriFragment' = $uriFragment
+        'Method' = 'Get'
+        'Description' =  $description
+        'AcceptHeader' = 'application/vnd.github.symmetra-preview+json'
+        'AccessToken' = $AccessToken
+        'TelemetryEventName' = $MyInvocation.MyCommand.Name
+        'TelemetryProperties' = $telemetryProperties
+        'NoStatus' = (Resolve-ParameterWithDefaultConfigurationValue -Name NoStatus -ConfigValueName DefaultNoStatus)
+    }
+
+    try
+    {
+        return Invoke-GHRestMethod @params
+    }
+    catch
+    {
+        Write-InteractiveHost "No reference named $Reference exists in repository $RepositoryName" -NoNewline -f Red
+        Write-Log -Level Error "No reference named $Reference exists in repository $RepositoryName"
+        return $null
+    }
+
+}
+
+function New-GitHubReference
+{
+    <#
+    .SYNOPSIS
+        Create a reference in a given GitHub repository.
+
+    .DESCRIPTION
+        Create a reference in a given GitHub repository.
+        The Git repo for this module can be found here: http://aka.ms/PowerShellForGitHub
+
+    .PARAMETER OwnerName
+        Owner of the repository.
+        If not supplied here, the DefaultOwnerName configuration property value will be used.
+
+    .PARAMETER RepositoryName
+        Name of the repository.
+        If not supplied here, the DefaultRepositoryName configuration property value will be used.
+
+    .PARAMETER Uri
+        Uri for the repository.
+        The OwnerName and RepositoryName will be extracted from here instead of needing to provide
+        them individually.
+
+    .PARAMETER Reference
+        The name of the fully qualified reference to be created (eg: refs/heads/master)
+
+    .PARAMETER Sha
+        The SHA1 value for the reference to be created
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .PARAMETER NoStatus
+        If this switch is specified, long-running commands will run on the main thread
+        with no commandline status update.  When not specified, those commands run in
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .EXAMPLE
+        -GitHubReference -OwnerName Powershell -RepositoryName PowerShellForGitHub -Reference refs/heads/master -Sha aa218f56b14c9653891f9e74264a383fa43fefbd

```

Comment:
Since the time that your PR started, this repo has moved from the `PowerShell` org to the `microsoft` org...

```suggestion
        -GitHubReference -OwnerName microsoft -RepositoryName PowerShellForGitHub -Reference refs/heads/master -Sha aa218f56b14c9653891f9e74264a383fa43fefbd
```